src/smart_account.ts

```diff
@@ -0,0 +1,14 @@
+import type { Contract } from 'ethers';
+
+export interface SmartAccountManager {
+  getAddress: (address?: string, salt?: number) => Promise<string>;
```

Comment:
What would passing in the optional `address` be used for?

Reply:
the smart contract address is generated using an address + salt, so in our case we'd default to using the current user eoa [eoa at slot 0 on mnemonic 0], but if we start to support many EOAs and mnemonics might be nice to allow for smart contract wallet generation derived from another EOA